src/bin/huak/commands/new.rs

```diff
@@ -42,5 +42,35 @@ pub fn run(path: String, app: bool, lib: bool) -> CliResult<()> {
     ops::new::create_project(&project)
         .map_err(|e| CliError::new(e, ExitCode::FAILURE))?;
 
+    if !no_vcs {
+        ops::new::init_vcs(&project)
+            .map_err(|e| CliError::new(e, ExitCode::FAILURE))?;
+    }
+
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use tempfile::tempdir;
+
+    #[test]
+    fn initializes_git() {
+        let directory = tempdir().unwrap().into_path();
+        std::env::set_current_dir(&directory).unwrap();
+        super::run("foo".to_owned(), true, false, false).unwrap();
+        let proj_dir = directory.join("foo");
+        assert!(proj_dir.is_dir());
+        assert!(proj_dir.join(".git").is_dir());
+    }
+
+    #[test]
+    fn does_not_initialize_git() {
+        let directory = tempdir().unwrap().into_path();
+        std::env::set_current_dir(&directory).unwrap();
+        super::run("foo".to_owned(), true, false, true).unwrap();
+        let proj_dir = directory.join("foo");
+        assert!(proj_dir.is_dir());
+        assert!(!proj_dir.join(".git").exists());
+    }
+}
```

Comment:
We should test in the ops space.

Reply:
The issue here is that we *can* test `init_vcs` from ops space, but we can't test there whether the argument passed to `command::new::run` is being correctly used, which is what these tests test (in addition to `init_vcs`, but that's a side effect).


And I'm not sure what needs to be tested and what doesn't. Personally I'm not a big fan of unit testing, especially granular/specific ones. So, for me, a more general test that covers a bigger part of the functionality is good enough. But to each their own. And I don't mind writing the tests themselves, it's just that I don't *feel* where the coverage line should be drawn.

Reply:
> but we can't test there whether the argument passed to command::new::run is being correctly used

This is a separate issue that we haven't solved yet for every other command as well. The tradeoff is manageable while the CLI app logic is super light. I've tinkered with testing through `clap` usage that I'd like to introduce here at some point. That's when I think testing in the bin space will be appropriate.

> Personally I'm not a big fan of unit testing, especially granular/specific ones

Hmm interesting. I'm the opposite. I like tons of unit-level coverage.

> it's just that I don't feel where the coverage line should be drawn.

I personally don't see an issue with over-optimizing for more coverage (making the tradeoff of redundancy if needed). But I don't think we're at a point where we need to worry about any redundant testing.

Reply:
I would compare premature testing to premature optimization. Unless it's some critical part, you probably don't need it tested. In this situation, even if something went wrong, worse case scenario is a git repo is not initialized. Which would be pretty easy to notice and then to debug. 
IMO it's a waste of developers' time, when writing tests and then when refactoring (and not even including conversations like this :P ). But then again, of all the things to "waste" time on, tests are not so bad.

Reply:
IMO requiring contributions to be paired with sufficient testing isn't more than what should be expected. I don't believe tests are ever premature. The earlier on the simple the tests (in theory). If what you are contributing is simple but requires significant effort to test, then I'd reconsider the implementation.

It's, imo, similar to choosing to using a more complicated programming language like Rust. You make an investment early on that feels like it's time potentially better spent elsewhere, but the ROI is worth it since that investment pays off with better stability, more correct code, etc..

It's roughly the same idea. The tests may seem like overkill today, but as we start adding more and more in, we'll be happy we have that coverage when we want to sustain our pace of development without breaking existing logic. Whether or not the coverage is related, the piece of mind alone is a huge win.

Reply:
Also I'm a little confused how this is related to moving the tests to `ops::new` 😅

Reply:
We went on a bit of a tangent here.

> Also I'm a little confused how this is related to moving the tests to `ops::new`

Barely at all. I just haven't been at my workstation to do the change until now. Done now.
  
  
 ---
To end the discussion: We have to draw a line somewhere. I would draw it in a different spot, but I accept where you draw it.